packages/connectors-web/src/alchemySmartWallet.ts

```diff
@@ -1,85 +1,233 @@
+  baseConnector: CreateConnectorFn;
+  /** API key for Alchemy authentication */
+  apiKey?: string;
+  /** JWT token for authentication */
+  jwt?: string;
+  /** Custom RPC URL (optional - defaults to Alchemy's API URL, but can be used to override the chain's Alchemy URL) */
+  url?: string;
 };
 
 alchemySmartWallet.type = "alchemy-smart-wallet" as const;
 
-type Provider = any; // TODO: Define proper provider type
+type Provider = SmartWalletClientEip1193Provider;
 type Properties = {
   readonly baseConnector: ReturnType<CreateConnectorFn>;
 };
 
-export function alchemySmartWallet({
-  createBaseConnector,
-}: AlchemySmartWalletOptions): CreateConnectorFn<Provider, Properties> {
+/**
+ * Creates a Wagmi connector that uses Alchemy's Wallet API to perform actions
+ * using a smart account. It wraps a base connector and uses it to sign
+ * transactions.
+ *
+ * @param {AlchemySmartWalletOptions} options - The options for the Alchemy Smart Wallet connector.
+ * @param {CreateConnectorFn} options.baseConnector - The base connector to use to sign transactions.
+ * @param {string} options.apiKey - The API key to use to authenticate with Alchemy.
+ * @param {string} options.jwt - The JWT to use to authenticate with Alchemy.
+ * @param {string} options.url - Custom RPC URL (optional - defaults to Alchemy's API URL, but can be used to override the chain's Alchemy URL)
+ * @returns {CreateConnectorFn} The Alchemy Smart Wallet connector, which can be passed to Wagmi's `createConfig`.
+ */
+export function alchemySmartWallet(
+  options: AlchemySmartWalletOptions,
+): CreateConnectorFn<Provider, Properties> {
   return createConnector((config) => {
-    const baseConnector = createBaseConnector(config);
+    // `config` contains an event emitter that all of Wagmi listens to. Don't
+    // let the inner connecter get its hands on it: we want to fully encapsulate
+    // it and not let it affect global state. Note that the uid doesn't matter
+    // since we're only using this internally.
+    //
+    // TODO: Listen for inner connect and disconnect events and update state accordingly.
+    const innerEmitter = new Emitter("inner-uid");
+    const innerConnector = options.baseConnector({
+      ...config,
+      emitter: innerEmitter,
+    });
+
+    /*
+     * We'll return this useless provider when the smart wallet connector is not
+     * connected. We need to return _something_ in response to `getProvider`
+     * even when not connected, or Wagmi will skip over this connector.
+     *
+     * TODO: Should we return the same provider instance every time, with that
+     * instance being a proxy object routing to either this or the real provider
+     * if it exists? We would need to do that if Wagmi held on to a provider
+     * that it first got during initialization. It doesn't seem to do that
+     * today, but what if it changed behavior?
+     */
+    const loggedOutProvider: Provider = {
+      on: () => {},
+      removeListener: () => {},
```

Comment:
Not sure if this is overkill/needed, do you think we should have some code to remove the listeners?

```
const listeners = new Map();
const loggedOutProvider: Provider = {
  on: (event, listener) => {
    if (!listeners.has(event)) listeners.set(event, new Set());
    listeners.get(event)?.add(listener);
    return () => listeners.get(event)?.delete(listener);
  },
  removeListener: (event, listener) => {
    listeners.get(event)?.delete(listener);
  },
};
```

Reply:
Since right now this provider never emits events, what I wrote should be equivalent behavior and I think there's no reason to store listeners that won't be called.

(We could forward listeners to the inner provider, but that feels like it could have unpredictable consequences.)

Reply:
`reconnect` awaits the provider: https://github.com/wevm/wagmi/blob/9651e18c3d140e37e16dcc80aabc29624344b7c5/packages/core/src/actions/reconnect.ts#L68

Reply:
the only time we would need the dummy provider during this implementation is during `reconnect`, right?

when `getProvider` is called here, could we await the inner connector being ready, then await the smart wallet client creation so that we have the provider here?